aten/src/ATen/HammerBladeGenerator.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <ATen/core/MT19937RNGEngine.h>
+
+namespace at {
+
+struct CAFFE2_API HammerBladeGenerator : public Generator {
```

Comment:
Just curious, which kernel needed this random number generator?

Reply:
Never mind, found this below. For initializing nn model parameters?

Reply:
actually the first use case is in `nn.Dropout`, which calls `bernoulli_`